x-pack/plugins/observability_solution/inventory/public/hooks/use_detail_view_redirect.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  ASSET_DETAILS_LOCATOR_ID,
+  AssetDetailsLocatorParams,
+  ENTITY_IDENTITY_FIELDS,
+  ENTITY_TYPE,
+  ENTITY_TYPES,
+  SERVICE_ENVIRONMENT,
+  SERVICE_OVERVIEW_LOCATOR_ID,
+  ServiceOverviewParams,
+} from '@kbn/observability-shared-plugin/common';
+import { useCallback } from 'react';
+import { DashboardLocatorParams } from '@kbn/dashboard-plugin/public';
+import { DASHBOARD_APP_LOCATOR } from '@kbn/deeplinks-analytics';
+import { castArray } from 'lodash';
+import { Exception } from 'handlebars';
+import type { Entity } from '../../common/entities';
+import { unflattenEntity } from '../../common/utils/unflatten_entity';
+import { useKibana } from './use_kibana';
+
+const KUBERNETES_DASHBOARDS_IDS: Record<string, string> = {
```

Comment:
This approach won't work for other spaces. As the integration package installs the dashboards for the default one (if I'm not mistaken).

Reply:
Good point. It won't indeed. I think users will have to do [this](https://www.elastic.co/guide/en/kibana/current/managing-saved-objects.html#managing-saved-objects-share-to-space) . @roshan-elastic , since we're delegating to the dashboards page the responsibility for checking the existence of a dashboard, should we check with them if the can create better messaging for troubleshooting?

Reply:
Hey @crespocarlos @kpatticha good thought.

We can go ahead with our part and I'll mention to @teresaalvarezsoler (Dashboards) and @daniela-elastic  (Integrations) about whether there might be a more helpful empty state.

To confirm, the problem is:

- We'll send a user to a dashboard by ID (e.g. https://edge-oblt.kb.us-west2.gcp.elastic-cloud.com/app/dashboards#/view/adsad)
- The dashboard may or may not exist depending on which space a user is in (because integrations install in the default space - not all spaces)
- If a user clicks through from a link in our UI to a dashboard and it doesn't exist - they'll hit this page:

![image](https://github.com/user-attachments/assets/973b484f-e8c2-4c20-be22-4cc98624a360)

We'd like to propose the dashboard empty state should handle 'missing dashboards' potentially a bit more gracefully (e.g. a link to common reasons why it may not show) or maybe there's something better we could do (e.g. pass through a query-string explaining we're expecting the dashboard to exist and have a toast pop up to suggest a more helpful call to action to fix it).

Is that right @crespocarlos?

cc @teresaalvarezsoler @daniela-elastic - we want to send users through to dashboards for some entities within Inventory (we'd start with K8s entities - for which we have dashboards which are packaged in the K8s integration). The problem we believe is that the dashboards only exist for the space you installed the integration in:

<img width="1269" alt="image" src="https://github.com/user-attachments/assets/a27db61a-7aa4-4ec2-9e90-45154f155b24">

FYI - All of this is 'tech preview' so we're not too worried right now

Reply:
> We'd like to propose the dashboard empty state should handle 'missing dashboards' potentially a bit more gracefully (e.g. a link to common reasons why it may not show) or maybe there's something better we could do (e.g. pass through a query-string explaining we're expecting the dashboard to exist and have a toast pop up to suggest a more helpful call to action to fix it).

@roshan-elastic yeah, something like that. If a dashboard id is found on any space, the empty state message could point users to a possible solution to the problem.

cc: @kpatticha

Reply:
Cool - @teresaalvarezsoler @daniela-elastic - I can mention in our next 1-2-1...more of an FYI for now